book.rb

```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.book = self
+  end
```

Comment:
- [ ] In order to implement the many-to-many and has-many-through relationship between the Book, Person, and Rental classes, please note that in the `add_rental` method, you can instantiate the `Rental` class and pass in the book, date, and person as parameters. This will automatically cause the `Rental` class in its initialize method to add itself to the `@rentals` array on both the book and person classes.
Please take a look at this [lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) for a more detailed explanation
**HINT: 👇**
### book.rb
```ruby
def add_rental(person, date)
    Rental.new(date, person, self)
end
```